dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/ToolErrorHandlingTools.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.fat.tool.basicToolApp;
+
+import java.io.IOException;
+
+import io.openliberty.mcp.annotations.Tool;
+import io.openliberty.mcp.annotations.ToolArg;
+import io.openliberty.mcp.tools.CustomBusinessException;
+import io.openliberty.mcp.tools.ToolCallException;
+import io.openliberty.mcp.tools.UnwrappedBusinessException;
+import io.openliberty.mcp.tools.WrapBusinessException;
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ *
+ */
+@ApplicationScoped
+public class ToolErrorHandlingTools {
+
+    //////////
+    // Exception Handling Tools
+    //////////
+
+    @Tool(name = "businessErrorTool", title = "Business Error Handler", description = "This tool throws error on business tool")
+    public String businessErrorTool(@ToolArg(name = "input", description = "Handles business tool error") String input) {
+        throw new ToolCallException("Invalid business input: " + input);
+    }
+
+    @Tool(name = "unwrappedExceptionTool", title = "Unwrapped Exception Tool", description = "Throws unwrapped RuntimeException")
+    public String rawExceptionTOol(@ToolArg(name = "input", description = "Throws raw exception") String input) {
+        throw new IllegalStateException("Internal server error");
+    }
```

Comment:
This test is for an exception thrown by user code, so they wouldn't write "Internal server error"

Since the exception is not `ToolCallException` and there's no `@WrapBusinessError` on the method, the response back to the client should still say `InternalServerError` and the message from the user's exception should be logged.

```suggestion
    @Tool(name = "unwrappedExceptionTool", title = "Unwrapped Exception Tool", description = "Throws unwrapped RuntimeException")
    public String rawExceptionTool(@ToolArg(name = "input", description = "Throws raw exception") String input) {
        throw new IllegalStateException("rawExceptionTool test error for " + input);
    }
```